crates/ruff/src/args.rs

```diff
@@ -282,7 +291,7 @@ pub struct CheckCommand {
     #[arg(short, long, env = "RUFF_NO_CACHE", help_heading = "Miscellaneous")]
     pub no_cache: bool,
     /// Ignore all configuration files.
-    #[arg(long, conflicts_with = "config", help_heading = "Miscellaneous")]
+    #[arg(long, help_heading = "Miscellaneous")]
```

Comment:
Yeah, `--isolated` still conflicts if `--config` is a file:

![image](https://github.com/astral-sh/ruff/assets/66076021/3af739ca-2ad2-4a46-87e3-b119b3e62123)

However, we now produce this error manually in `ConfigArgs.from_cli_options`, rather than using clap's out-of-the-box mechanism. The reason is that at this stage, we don't yet know whether the `--config` flag is being used to specify a config file (which _does_ conflict with `--isolated`) or whether it's being used to specify a config override (which _doesn't_ conflict with `--isolated`).

I'll add a clarifying comment.